test/test_log.cpp

```diff
@@ -89,6 +89,14 @@ BOOST_AUTO_TEST_CASE(strings_are_serialized_with_double_quotes)
     BOOST_CHECK_EQUAL("\"string\"", to_string(std::string("string")));
 }
 
+BOOST_AUTO_TEST_CASE(null_c_strings_are_serialized_to_nullptr)
+{
+    const char* const null_str = nullptr;
+    BOOST_CHECK_EQUAL("nullptr", to_string(null_str));
+    // Note the lack of double quotes in the output, as opposed to when the
+    // string "nullptr" is serialized.
```

Comment:
nit: I’m not sure we need that comment, do we?
If we really need to emphasize this, what about
```suggestion
    BOOST_CHECK_NE("\"nullptr\"", to_string(null_str));
```
?
That’s a little odd following an `BOOST_CHECK_EQUAL` though, I suppose 😅

Reply:
How about adding `to_string("nullptr")` as a `BOOST_CHECK_EQUAL` right below instead of the comment to demonstrate it instead of explaining it?

Reply:
Yes, demonstrating the difference rather than attempting to explain it makes a lot of sense and has the advantage of proving it to be the case into the bargain.